quarkus/deployment/src/main/java/org/keycloak/quarkus/deployment/KeycloakProcessor.java

```diff
@@ -782,31 +781,89 @@ private Map<String, ProviderFactory> loadDeployedScriptProviders(ClassLoader cla
         return providers;
     }
 
-    private ProviderFactory createDeployableScriptProvider(JarFile jarFile, Entry<String, List<ScriptProviderMetadata>> entry,
-            ScriptProviderMetadata metadata) throws IOException {
-        String fileName = metadata.getFileName();
+    private List<ScriptProviderDescriptor> getScriptProviderDescriptorsFromClassPath(URL url) throws IOException {
+        String file = url.getFile();
+
+        if (!file.endsWith(".json")) {
+            return List.of();
+        }
+
+        List<ScriptProviderDescriptor> descriptors = new ArrayList<>();
+
+        try (InputStream is = url.openStream()) {
+            ScriptProviderDescriptor descriptor = JsonSerialization.readValue(is, ScriptProviderDescriptor.class);
 
-        if (fileName == null) {
-            throw new RuntimeException("You must provide the script file name");
+            configureScriptDescriptor(descriptor, fileName -> {
+                // descriptor is at META-INF/
+                Path basePath = Path.of(url.getPath()).getParent().getParent();
```

Comment:
Just wondering about avoiding these walks up in the path hierarchy and how stable it is. ATM, the scripts are in the same dir, but it can change in some case, right? However, I guess it's not a big problem.

Reply:
AFAIK, scripts are always in the root path of the classpath where the descriptor is located.

Reply:
ok, makes sense